frontend/src/component/common/Combo/Combo.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+    HTMLAttributes,
+    forwardRef,
+    useCallback,
+    useRef,
+    useState,
+} from 'react';
+import { Autocomplete, AutocompleteProps, styled } from '@mui/material';
+import { GroupedVirtuoso, GroupedVirtuosoHandle } from 'react-virtuoso';
+
+const VIRTUALIZED_THRESHOLD = 100;
+
+const StyledDiv = styled('div')(({ theme }) => ({
+    'div[data-item-group-index]': {
+        '.MuiAutocomplete-option': {
+            paddingLeft: theme.spacing(3),
+        },
+    },
```

Comment:
Tried adding `overflow: 'hidden'` or `padding: 0` here to improve UI/UX and avoid some annoying extra scrolling, but somehow it made scrolling the dropdown more sluggish? Maybe it's a compromise we want, but needs more exploring.